src/link_manager/LinkManagerStateMachineActiveActive.cpp

```diff
@@ -0,0 +1,934 @@
+            break;
+        case mux_state::MuxState::Label::Standby:
+            enterLinkProberState(compositeState, link_prober::LinkProberState::Label::Unknown);
+            break;
+        case mux_state::MuxState::Label::Unknown:
+            enterLinkProberState(compositeState, link_prober::LinkProberState::Label::Wait);
+            break;
+        case mux_state::MuxState::Label::Error:
+            enterLinkProberState(compositeState, link_prober::LinkProberState::Label::Wait);
+            break;
+        case mux_state::MuxState::Label::Wait:
+            enterLinkProberState(compositeState, link_prober::LinkProberState::Label::Wait);
+            break;
+        default:
+            break;
+    }
+}
+
+//
+// ---> initPeerLinkProberState();
+//
+// initialize peer LinkProberState when configuring the composite state machine
+//
+void ActiveActiveStateMachine::initPeerLinkProberState()
+{
+    switch (mPeerMuxState) {
+        case mux_state::MuxState::Label::Active:
+            enterPeerLinkProberState(link_prober::LinkProberState::Label::PeerActive);
+            break;
+        case mux_state::MuxState::Label::Standby:
+            enterPeerLinkProberState(link_prober::LinkProberState::Label::PeerUnknown);
+            break;
+        case mux_state::MuxState::Label::Unknown:
+            enterPeerLinkProberState(link_prober::LinkProberState::Label::PeerWait);
+            break;
+        case mux_state::MuxState::Label::Error:
+            enterPeerLinkProberState(link_prober::LinkProberState::Label::PeerWait);
+            break;
+        case mux_state::MuxState::Label::Wait:
+            enterPeerLinkProberState(link_prober::LinkProberState::Label::PeerWait);
+            break;
+        default:
+            break;
+    }
+}
+
+//
+// ---> startMuxProbeTimer();
+//
+// start a mux probe and wait for mux probe notification(active or standby) from xcvrd
+//
+void ActiveActiveStateMachine::startMuxProbeTimer()
+{
+    probeMuxState();
+    mDeadlineTimer.expires_from_now(boost::posix_time::milliseconds(
+        mMuxProbeBackoffFactor * mMuxPortConfig.getNegativeStateChangeRetryCount() * mMuxPortConfig.getTimeoutIpv4_msec()
+    ));
+    mDeadlineTimer.async_wait(getStrand().wrap(boost::bind(
+        &ActiveActiveStateMachine::handleMuxProbeTimeout,
+        this,
+        boost::asio::placeholders::error
+    )));
+}
+
+//
+// ---> handleMuxProbeTimeout(boost::system::error_code errorCode);
+//
+// handles when xcvrd has timeout responding mux probe
+//
+void ActiveActiveStateMachine::handleMuxProbeTimeout(boost::system::error_code errorCode)
+{
+    MUXLOGDEBUG(mMuxPortConfig.getPortName());
+    if (errorCode == boost::system::errc::success) {
+        if (ms(mCompositeState) == mux_state::MuxState::Label::Unknown ||
+            ms(mCompositeState) == mux_state::MuxState::Label::Error ||
+            ms(mCompositeState) == mux_state::MuxState::Label::Wait) {
```

Comment:
Thought we got rid of `Wait` in active-active?

Reply:
Yes, `Wait` will only occur in the initial state `(Wait, Wait, Down)`